sklearn/utils/extmath.py

```diff
@@ -699,7 +702,10 @@ def _incremental_mean_and_var(X, last_mean=.0, last_variance=None,
     if last_variance is None:
         updated_variance = None
     else:
-        new_unnormalized_variance = X.var(axis=0) * new_sample_count
+        if accepting_nan is True:
+            new_unnormalized_variance = np.nanvar(X, axis=0) * new_sample_count
+        else:
+            new_unnormalized_variance = X.var(axis=0) * new_sample_count
```

Comment:
The new sample count also needs updating, and will be different for different features.

Reply:
Given that tests are currently passing, I think we can safely say that the tests are not correct or are not strong enough yet.